tests/integration-tests/CMakeLists.txt

```diff
@@ -55,10 +55,13 @@ target_link_libraries(
   mir-test-doubles-static
 
   mircommon
+  miral
+  miral-internal
```

Comment:
Linking both `miral` (which includes `miral-internal` internally) and `miral-internal` risk multiple definitions. (Which is at best non-portable, and at worst causes hard to track down bugs.)